testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -5167,6 +5167,18 @@ private static void checkIf(SqlOperatorFixture f) {
     f.checkNull("array_concat(cast(null as integer array), array[1])");
   }
 
+  /** Tests {@code ARRAY_DISTINCT} function from Spark. */
+  @Test void testArrayDistinctFunc() {
+    SqlOperatorFixture f = fixture()
+        .setFor(SqlLibraryOperators.ARRAY_DISTINCT)
+        .withLibrary(SqlLibrary.SPARK);
+    f.checkScalar("array_distinct(array[1, 2, 2, 1])", "[1, 2]",
+        "INTEGER NOT NULL ARRAY NOT NULL");
+    f.checkScalar("array_distinct(array[null, 1, null])", "[null, 1]",
+        "INTEGER ARRAY NOT NULL");
+    f.checkNull("array_distinct(null)");
+  }
```

Comment:
missing happy path:

```suggestion
    f.checkScalar("array_distinct(array[1, 2])", "[1, 2]",
        "INTEGER NOT NULL ARRAY NOT NULL");

    f.checkScalar("array_distinct(array[1])", "[1]",
        "INTEGER NOT NULL ARRAY NOT NULL");
  }
```

Reply:
why need this, it is the same. unit tests need to be carefully selected

Reply:
`array[1, 2]`: is a simpler form, if somethings is wrong with more complex forms (eg `array[1, 2, null]`) it's helpful to debug.
`array[1]`: it's helpful to check for 1 off errors in the return type inference

Reply:
How about `array_distinct(array[0, 0.0])`?

Should we need unit tests for array<varchar>?

Reply:
don't need. array should be element with same type, and it doesn't support implicit cast. so i guess it will throw exception.